### PR TITLE
Event creation email

### PR DESCRIPTION
**Pull request by Mathieu414**

### CODE REVIEW

File: engine/api/mailer/MockMailer.php [hunk 1]
```diff
@@ -8,20 +8,32 @@ class MockMailer extends Mailer
...
         $this->mockedMail->subject = $subject;
         $this->mockedMail->content = $content;
         return $this;
     }
 
+    function createBulkEmails($addresses, $subject, $content): self
+    {
+        foreach ($addresses as $address => $name) {
+            array_push($this->mockedMail->addresses, $address);
+        }
```
Comment (Atmos4) [inline engine/api/mailer/MockMailer.php:27]: You can just directly assign here I think, or use array_merge

```suggestion
        $this->mockedMail->addresses = $addresses;
```

File: engine/api/mailer/Mailer.php [hunk 2]
```diff
@@ -92,4 +107,13 @@ static function createActivationEmail(string $address, string $token)
         $content = "Voici le lien pour activer ton compte: $base_url/activation?token=$token";
         return Mailer::create()->createEmail($address, $subject, $content);
     }
+
+    static function createEventPublicationEmail(Event $event)
+    {
+        $base_url = env("BASE_URL");
+        $subject = "Nouvel événement sur l'intranose";
+        $content = "Un nouvel événement a été publié sur l'intranose ! Pour aller le voir : <a href = '$base_url/evenements/$event->id' >Nouvel événement</a>";
```
Comment (Atmos4) [inline engine/api/mailer/Mailer.php:115]: Maybe include the name of the event somewhere? You could also add the deadline?

File: app/pages/events/event_publish.php [hunk 3]
```diff
@@ -11,10 +11,13 @@
     $event->open = !$event->open;
     em()->persist($event);
     em()->flush();
-    redirect("/evenements/$event->id");
+    if ($event->open && isset($_POST['send_email'])) {
+        MailerFactory::createEventPublicationEmail($event)->send();
+    }
+    #redirect("/evenements/$event->id");
```
Comment (Atmos4) [inline app/pages/events/event_publish.php:17]: Uncomment this?

File: engine/api/mailer/MockMailer.php (hunk 1 above)
Comment (Mathieu414) [reply engine/api/mailer/MockMailer.php:27]: Just did that, thx

File: engine/api/mailer/Mailer.php (hunk 2 above)
Comment (Mathieu414) [reply engine/api/mailer/Mailer.php:115]: Good one, I just added some more informations there 🤝

File: app/pages/events/event_publish.php (hunk 3 above)
Comment (Mathieu414) [reply app/pages/events/event_publish.php:17]: 🤦‍♂️

**Review by Atmos4 — APPROVED** [review]

**Pull request merged**